apps/tlon-mobile/src/hooks/useNotificationListener.ts

```diff
@@ -0,0 +1,69 @@
+import type { NavigationProp } from '@react-navigation/native';
+import { useNavigation } from '@react-navigation/native';
+import { parseActiveTab } from '@tloncorp/shared';
+import { addNotificationResponseReceivedListener } from 'expo-notifications';
+import { useEffect, useState } from 'react';
+
+import { useWebViewContext } from '../contexts/webview/webview';
+import { markChatRead } from '../lib/chatApi';
+import { connectNotifications } from '../lib/notifications';
+import type { TabParamList } from '../types';
+import { getPathFromWer } from '../utils/string';
+
+export default function useNotificationListener(initialWer?: string) {
+  const navigation = useNavigation<NavigationProp<TabParamList>>();
+  const webviewContext = useWebViewContext();
+  const [gotoPath, setGotoPath] = useState(
+    initialWer ? getPathFromWer(initialWer) : ''
+  );
+
+  // Start notifications prompt
+  useEffect(() => {
+    connectNotifications();
+  }, []);
+
+  useEffect(() => {
+    // Start notification tap listener
+    // This only seems to get triggered on iOS. Android handles the tap and other intents in native code.
+    const notificationTapListener = addNotificationResponseReceivedListener(
+      (response) => {
+        const {
+          actionIdentifier,
+          userText,
+          notification: {
+            request: {
+              content: { data },
+            },
+          },
+        } = response;
+        if (actionIdentifier === 'markAsRead' && data.channelId) {
+          markChatRead(data.channelId);
+        } else if (actionIdentifier === 'reply' && userText) {
+          // TODO: Send reply
+        } else if (data.wer) {
+          setGotoPath(getPathFromWer(data.wer));
```

Comment:
We should extract the path from the `wer` when it's initially received so we only need to do it once.